src/lib/message/L1ToL2Message.ts

```diff
@@ -553,6 +554,42 @@ export class L1ToL2MessageReaderClassic {
       ])
     )
   }
+
+  /**
+   * Try to get the receipt for the retryable ticket creation.
+   * This is the L2 transaction that creates the retryable ticket.
+   * If confirmations or timeout is provided, this will wait for the ticket to be created
+   * @returns Null if retryable has not been created
+   */
+  public async getRetryableCreationReceipt(
+    confirmations?: number,
+    timeout?: number
+  ): Promise<TransactionReceipt | null> {
+    if (!this.retryableCreationReceipt) {
+      this.retryableCreationReceipt = await getTransactionReceipt(
+        this.l2Provider,
+        this.retryableCreationId,
+        confirmations,
+        timeout
+      )
+    }
+
+    return this.retryableCreationReceipt || null
+  }
+
+  public async getMessageStatus(): Promise<L1ToL2MessageStatus> {
+    const creationReceipt = await this.getRetryableCreationReceipt()
+
+    if (!isDefined(creationReceipt)) {
+      return L1ToL2MessageStatus.NOT_YET_CREATED
+    }
+
+    if (creationReceipt.status === 0) {
+      return L1ToL2MessageStatus.CREATION_FAILED
+    }
+
+    return L1ToL2MessageStatus.REDEEMED
```

Comment:
Do we need to discern between expired and redeemed? We derive the l2txHash from the retryablecreationid:
```
private static calculateL2DerivedHash(
    retryableCreationId: string,
    l2TxnType: L2TxnType
  ): string {
    return keccak256(
      concat([
        zeroPad(retryableCreationId, 32),
        zeroPad(BigNumber.from(l2TxnType).toHexString(), 32),
      ])
    )
  }
```
then check if that tx was successful:
```
    if (l2TxReceipt && l2TxReceipt.status === 1) {
      return L1ToL2MessageStatus.REDEEMED
    }
```
otherwise the ticket would be EXPIRED.

Reply:
@yahgwai Is `l2TxnType` taken from `creationReceipt.type`? I'm using it to call `calculateL2DerivedHash`, and then running `l2Provider.getTransactionReceipt(
      l2DerivedHash
    )` but it's returning null